crates/connectors/ndc-postgres/src/configuration.rs

```diff
@@ -25,54 +77,232 @@ pub const CURRENT_VERSION: u32 = 1;
+            pool_settings: v1_config.pool_settings.clone(),
+            connection_uri: match &v1_config.connection_uri {
+                ConnectionUri::Uri(ResolvedSecret(uri)) => uri.clone(),
+            },
+        },
+        RawConfiguration::Version2(v2_config) => RuntimeConfiguration {
+            metadata: v2_config.metadata.clone(),
+            pool_settings: v2_config.pool_settings.clone(),
+            connection_uri: match &v2_config.connection_uri {
+                ConnectionUri::Uri(ResolvedSecret(uri)) => uri.clone(),
+            },
+        },
+    }
+}
+
+// for tests
+
+pub fn set_connection_uri(config: RawConfiguration, connection_uri: String) -> RawConfiguration {
+    match config {
+        RawConfiguration::Version1(v1) => RawConfiguration::Version1(version1::RawConfiguration {
+            connection_uri: ConnectionUri::Uri(ResolvedSecret(connection_uri)),
+            ..v1
+        }),
+        RawConfiguration::Version2(v2) => RawConfiguration::Version2(version2::RawConfiguration {
+            connection_uri: ConnectionUri::Uri(ResolvedSecret(connection_uri)),
+            ..v2
+        }),
+    }
 }
 
-impl<'a> version1::Configuration {
-    /// Apply the common interpretations on the Configuration API type into an RuntimeConfiguration.
-    pub fn as_runtime_configuration(self: &'a Configuration) -> RuntimeConfiguration {
-        RuntimeConfiguration {
-            metadata: metadata_to_current(&self.config.metadata),
+// The below exists solely to support indicating version 1 with an integer literal.
+// Once version 1 is phased out completely, the rest of this file is to be deleted and trait
+// implementations for Serialize/Deserialize/JsonSchema reverted to their derived versions.
+
+#[derive(Clone, Deserialize, Serialize)]
+pub struct RawConfigurationCompat(serde_json::Value);
+
+impl From<RawConfiguration> for RawConfigurationCompat {
+    fn from(value: RawConfiguration) -> Self {
+        let val = match value {
+            RawConfiguration::Version1(v1) => {
+                let mut val = serde_json::to_value(v1).unwrap();
+                let obj = val.as_object_mut().unwrap();
+
+                let mut res = serde_json::map::Map::new();
+                res.insert("version".to_string(), serde_json::json!(1));
+                res.append(obj);
+                serde_json::value::to_value(res).unwrap()
+            }
+            RawConfiguration::Version2(v2) => {
+                let mut val = serde_json::to_value(v2).unwrap();
+                let obj = val.as_object_mut().unwrap();
+
+                let mut res = serde_json::map::Map::new();
+                res.insert("version".to_string(), serde_json::json!("2"));
+                res.append(obj);
+                serde_json::value::to_value(res).unwrap()
+            }
+        };
+
+        RawConfigurationCompat(val)
+    }
+}
+
+// Dump of derivied JsonSchema trait implementation, tweaked to put version as an integer for
```

Comment:
We need to remember this is here and needs updating otherwise we're going to have such a confusing time debugging JSON schema.

Reply:
If we are able to put the bad generated stuff in a separate file that'd be great for clarity but only if it's easy to do so.

Reply:
Done!